fflib/src/classes/fflib_QueryFactoryTest.cls

```diff
@@ -151,6 +151,45 @@ private class fflib_QueryFactoryTest {
 		System.assert( Pattern.matches('SELECT.*Email.*FROM.*',query), 'Expected Name field in query, got '+query);
 	}
 
+	@isTest
+	static void orderingSet(){
+		fflib_QueryFactory qf = new fflib_QueryFactory(Contact.SObjectType);
+		qf.selectField('name');
+		qf.selectField('email');
+		qf.setCondition( 'name = \'test\'' );
+		qf.setOrdering( new fflib_QueryFactory.Ordering('Contact','LastModifiedDate',fflib_QueryFactory.SortOrder.DESCENDING) );
+		
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
+
+		qf.setOrdering('LastModifiedDate', fflib_QueryFactory.SortOrder.DESCENDING, true);
+		
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
+
+		qf.setOrdering(Contact.LastModifiedDate, fflib_QueryFactory.SortOrder.DESCENDING, true);
+
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
+
+		qf.setOrdering('LastModifiedDate', fflib_QueryFactory.SortOrder.DESCENDING);
+
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
+
+		qf.setOrdering(Contact.LastModifiedDate, fflib_QueryFactory.SortOrder.DESCENDING);
+
+		System.assertEquals(1,qf.getOrderings().size());
+		System.assertEquals(Contact.LastModifiedDate,qf.getOrderings()[0].getField() );
+		System.assertEquals(fflib_QueryFactory.SortOrder.DESCENDING,qf.getOrderings()[0].getDirection() );
```

Comment:
**ACTION: Test different orderings, and specifically test applying ASCENDING sort order.**
Each setOrdering overload produces the same Ordering. I know it's unlikely and I admit I'm being picky, but theoretically the tests could be passing if the first setOrdering call sets it to `LastModifiedDate DESC` and the subsequent calls have no effect.
Each setOrdering should either create a different resulting order OR the test should be split out so that each setOrdering overload is tested in isolation.